cpp/daal/src/algorithms/linear_regression/oneapi/linear_regression_train_dense_normeq_helper_oneapi_impl.i

```diff
@@ -64,20 +66,30 @@ services::Status KernelHelperOneAPI<algorithmFPType>::copyBetaToResult(const ser
     const services::String options = getKeyFPType<algorithmFPType>();
     services::String cachekey("__daal_algorithms_linear_regression_training_helper_");
     cachekey.add(options);
+
     factory.build(ExecutionTargetIds::device, cachekey.c_str(), clKernelHelperBetaCopy, options.c_str(), status);
     DAAL_CHECK_STATUS_VAR(status);
 
     const char * const kernelName = "copyBeta";
     KernelPtr kernel              = factory.getKernel(kernelName, status);
     DAAL_CHECK_STATUS_VAR(status);
 
+    DAAL_ASSERT(nBetas <= services::internal::MaxVal<uint32_t>::get());
+    DAAL_ASSERT(nBetasIntercept <= services::internal::MaxVal<uint32_t>::get());
+    DAAL_ASSERT(intercept <= services::internal::MaxVal<uint32_t>::get());
+    DAAL_ASSERT(nResponses <= services::internal::MaxVal<uint32_t>::get());
+
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, nResponses, nBetasIntercept);
+    DAAL_ASSERT(betaTmp.size() >= nResponses * nBetasIntercept);
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, nResponses, nBetas);
+    DAAL_ASSERT(betaRes.size() >= nResponses * nBetas);
+
     KernelArguments args(5, status);
-    DAAL_CHECK_STATUS_VAR(status);
-    args.set(0, betaTmp, AccessModeIds::read);
-    args.set(1, nBetas);
-    args.set(2, nBetasIntercept);
-    args.set(3, betaRes, AccessModeIds::write);
-    args.set(4, intercept);
+    args.set(0, /*const __global algorithmFPType * src =*/betaTmp, AccessModeIds::read);
+    args.set(1, /*uint nCols =*/static_cast<uint32_t>(nBetas));
+    args.set(2, /*uint nColsSrcs =*/static_cast<uint32_t>(nBetasIntercept));
+    args.set(3, /*__global algorithmFPType * dst =*/betaRes, AccessModeIds::write);
+    args.set(4, /*uint intercept =*/static_cast<uint32_t>(intercept));
 
```

Comment:
What's reason of adding those comments? Why this isn't added in other `args.set` places?

Reply:
Cause in this particular place names of arguments are not same in kernel and on host. In other places names are matching

Reply:
Fixed